### PR TITLE
Create non-root user and install sudo and gosu

### PR DESCRIPTION
**Pull request by gqmelo**
This is an intermediate step to build packages as a non-root user.
Current recipes should still work with this image without changes.

### CODE REVIEW

File: linux-anvil/Dockerfile [hunk 1]
```diff
@@ -46,6 +47,20 @@ RUN curl -s -L -O https://github.com/krallin/tini/releases/download/v0.10.0/tini
...
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.9/gosu-amd64" && \
+    openssl sha256 /usr/local/bin/gosu | grep 5ec5d23079e94aea5f7ed92ee8a1a34bbf64c2d4053dadf383992908a2f9dc8a && \
+    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.9/gosu-amd64.asc" && \
+    gpg --verify /usr/local/bin/gosu.asc && \
+    rm /usr/local/bin/gosu.asc && \
+    chmod +x /usr/local/bin/gosu
+
+# give sudo permission for conda user to run yum (user creation is postponed
+# to the entrypoint, so we can create a user with the same id as the host)
+RUN echo 'conda ALL=NOPASSWD: /usr/bin/yum' >> /etc/sudoers
```
Comment (jakirkham) [inline linux-anvil/Dockerfile]: Maybe this should be `ALL=(ALL) NOPASSWD:ALL`.

**Comment by jakirkham** [comment]
As long as everyone is on board with the plan in PR ( https://github.com/conda-forge/docker-images/pull/36 ), we should be able to merge this in short order. This just sets things up for switching to a non-`root` user, but doesn't actually change anything. Please share your thoughts.

cc @conda-forge/core @rgbkrk @parente

**Comment by jakirkham** [comment]
This certainly doesn't have to happen now, but would be worth considering down the road. There is something called [`su-exec`]( https://github.com/ncopa/su-exec ), which is similar to [`gosu`]( https://github.com/tianon/gosu ). Main differences are it is C and it is much younger.

File: linux-anvil/Dockerfile [hunk 2]
```diff
@@ -46,6 +47,20 @@ RUN curl -s -L -O https://github.com/krallin/tini/releases/download/v0.10.0/tini
     chmod +x tini && \
     mv tini /usr/local/bin
 
+# Install gosu to be used in the entrypoint, reason why to use it instead of
+# su or sudo can be found on the project docs
+RUN gpg --keyserver ha.pool.sks-keyservers.net --recv-keys B42F6819007F00F88E364FD4036A9C25BF357DD4
+RUN curl -o /usr/local/bin/gosu -SL "https://github.com/tianon/gosu/releases/download/1.9/gosu-amd64" && \
+    openssl sha256 /usr/local/bin/gosu | grep 5ec5d23079e94aea5f7ed92ee8a1a34bbf64c2d4053dadf383992908a2f9dc8a && \
+    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.9/gosu-amd64.asc" && \
```
Comment (jakirkham) [inline linux-anvil/Dockerfile]: There is now `1.10`.

Comment (gqmelo) [inline linux-anvil/Dockerfile]: Thanks, I've updated it

**Comment by jakirkham** [comment]
So the `Dockerfile` changes I would love to merge ASAP. However, the `entrypoint` changes are still evolving a little bit. Could we please pull them out of this PR and back into the other one? Then I will happily merge this.

**Comment by gqmelo** [comment]
Ok, done!

**Comment by jakirkham** [comment]
Awesome! Thanks. Will merge once CI passes.

File: linux-anvil/Dockerfile [hunk 3]
```diff
@@ -46,6 +47,21 @@ RUN curl -s -L -O https://github.com/krallin/tini/releases/download/v0.10.0/tini
...
+    openssl sha256 /usr/local/bin/gosu | grep 5b3b03713a888cee84ecbf4582b21ac9fd46c3d935ff2d7ea25dd5055d302d3c && \
+    curl -o /usr/local/bin/gosu.asc -SL "https://github.com/tianon/gosu/releases/download/1.10/gosu-amd64.asc" && \
+    openssl sha256 /usr/local/bin/gosu.asc | grep b026f831a268f6c31daec4234547e4bb0228400babad3d6f6561b6626855af44 && \
+    gpg --verify /usr/local/bin/gosu.asc && \
+    rm /usr/local/bin/gosu.asc && \
+    chmod +x /usr/local/bin/gosu
+
+# give sudo permission for conda user to run yum (user creation is postponed
+# to the entrypoint, so we can create a user with the same id as the host)
+RUN echo 'conda ALL=NOPASSWD: /usr/bin/yum' >> /etc/sudoers
```
Comment (jakirkham) [inline linux-anvil/Dockerfile]: Even though we haven't created the user yet, this doesn't seem to cause a problem FWICT by doing some local testing.

**Comment by jakirkham** [comment]
Small wrinkle after deploy was the upload token was being written to a `root` owned directory. Just fixed that a few minutes ago.

**Pull request merged**